src/Sulu/Bundle/SecurityBundle/Entity/User.php

```diff
@@ -72,6 +82,21 @@ public function getUserRoles()
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function getRoles()
+    {
+        $roles = parent::getRoles();
+
+        foreach ($this->getUserRoles() as $userRole) {
+            /** @var UserRole $userRole */
+            $roles[] = $userRole->getRole()->getIdentifier();
+        }
```

Comment:
Hm, I can't imagine of any... And apart from that the `getRoles` method is defined in Symfony, so no chance to put a parameter there.